app/services/open_library_service.rb

```diff
@@ -1,17 +1,25 @@
 class OpenLibraryService
+  include HTTParty
+  base_uri 'https://openlibrary.org'
+
+  def initialize
+    @options = { query: { format: 'json', jscmd: 'data' } }
+  end
+
   def book_info(isbn)
-    url = 'https://openlibrary.org/api/books?bibkeys=ISBN:' + isbn + '&format=json&jscmd=data'
-    @data = (JSON.parse HTTParty.get(url).to_s).values[0]
-    @response = {}
-    generate_response unless @data.nil?
-    @response
+    @options[:query][:bibkeys] = "ISBN:#{isbn}"
+    data = (JSON.parse self.class.get('/api/books', @options).to_s).values[0]
+    generate_response(data)
   end
 
-  def generate_response
-    @response['isbn'] = @data['identifiers']['isbn_10'][0]
-    @response['title'] = @data['title']
-    @response['subtitle'] = @data['subtitle']
-    @response['nnumber_of_pages_pages'] = @data['number_of_pages']
-    @response['authors'] = @data['authors'].map { |author| author['name'] }
+  def generate_response(data)
+    return {} if data.blank?
+    response = {}
+    response['isbn'] = data['identifiers']['isbn_10'][0]
```

Comment:
It's just a matter of opinion (not a correction) but I prefer just returning a hash directly (check my previous correction) instead of doing something like
```
variable = {}
variable[key] = some_code
variable[other_key] = some_other code
variable
```